### PR TITLE
docs: Standardize the README

### PR DESCRIPTION
**Pull request by gnumoksha**
What:
Standardizes the README file providing a common language and an implementation link.

Why:
There are differences between PSR's READMEs in regarding language and the lacking of implementations references. So I've updated all READMEs using the https://github.com/php-fig/http-factory as base.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-This repository holds all interfaces related to [PSR-17 (HTTP Message Factories)][psr-17]. 
-Please refer to the specification for a description.
+This repository holds all interfaces related to [PSR-17 (HTTP Factories)][psr-url].
 
-You can find implementations of the specification by looking for packages providing the 
-[psr/http-factory-implementation](https://packagist.org/providers/psr/http-factory-implementation) virtual package.
+Note that this is not a HTTP Factory implementation of its own. It is merely interfaces that describe the components of a HTTP Factory.
 
-[psr-17]: https://www.php-fig.org/psr/psr-17/
+You can find [implementations][implementation-url] and [installation instructions][package-url] for the specification on the packagist.
```
Comment (Jean85) [inline README.md]: In here the reference to the virtual package is lost in the link

**Review by Jean85 — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (shadowhand) [reply README.md]:
```suggestion
The installable [package][package-url] and [implementations][implementation-url] are listed on Packagist.
```

File: README.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+
+[psr-url]: https://www.php-fig.org/psr/psr-17/
+[package-url]: https://packagist.org/packages/psr/http-factory
```
Comment (shadowhand) [inline README.md]:
```suggestion
```

**Review by shadowhand — CHANGES_REQUESTED** [review]

**Review by shadowhand — APPROVED** [review]

**Comment by shadowhand** [comment]
@Jean85 looks good to me now. 👍 ?

**Comment by gnumoksha** [comment]
ping

**Review by shadowhand — APPROVED** [review]

**Pull request merged**